### PR TITLE
Add armv8l support (Closes: #18898)

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Treat armv8l as arm64
Also fix typo "armv71"

### CODE REVIEW

**Comment by ghost** [comment]
Maybe also worth adding this to csourses_v1?

**Comment by FedericoCeratto** [comment]
@Yardanico csources_v1/build.sh ?

File: tools/niminst/makefile.nimf [hunk 1]
```diff
@@ -157,6 +157,9 @@ endif
 ifeq ($(ucpu),armv7hl)
   mycpu = arm
 endif
+ifeq ($(ucpu),armv8l)
+  mycpu = arm64
```
Comment (alaviss) [inline tools/niminst/makefile.nimf]: Are you sure that it should be treated like arm64? It's rather weird to see toolchain reporting `armv8l`, I'm not sure if this means its an armv8 cpu in 32-bit mode or 64-bit mode. From this issue https://github.com/randombit/botan/issues/1543#issuecomment-382186505 it looks like it should be 32-bit, but I'm not sure.

**Review by alaviss — CHANGES_REQUESTED** [review]
You should apply a similar change to `tools/niminst/buildsh.nimf` too

File: tools/niminst/makefile.nimf (hunk 1 above)
Comment (FedericoCeratto) [reply tools/niminst/makefile.nimf]: ARMv8 is native at 64bit but can also run in 32bit mode (for backward compatibility)

Comment (alaviss) [reply tools/niminst/makefile.nimf]: Yea but we are detecting using `uname -m` and `armv8l` seems to be what is returned when the kernel is 32-bit.

So I'm unsure if we should default to 64-bit here. Do toolchains running on these systems target aarch64?

Comment (FedericoCeratto) [reply tools/niminst/makefile.nimf]: The arm64 where I'm seeing the build fail is identifying as armv8l: https://buildd.debian.org/status/fetch.php?pkg=nim&arch=armhf&ver=1.4.8-3&stamp=1629381474&raw=0

Comment (alaviss) [reply tools/niminst/makefile.nimf]: Isn't that build targeting armhf, which is 32-bit?

Comment (FedericoCeratto) [reply tools/niminst/makefile.nimf]: It is, but I'm not sure if we wouldn't get armv8l also when targeting arm64 or not.

Comment (alaviss) [reply tools/niminst/makefile.nimf]: Given that 64bit arm is pretty much standardized as aarch64 everywhere, I think it's safe to assume that armv8l is marker for aarch32, esp when that's what debian yields

Comment (FedericoCeratto) [reply tools/niminst/makefile.nimf]: Right! Thank you! I updated the PR.
This helps: https://bgamari.github.io/posts/2019-06-12-arm-terminology.html

**Review by alaviss — APPROVED** [review]
LGTM, need to restart CI though

**Comment by timotheecour** [comment]
I've added `armv8l` to https://github.com/nim-lang/website/pull/301

**Pull request merged**